src/dotnet-core-uninstall/Shared/VSVersioning/VisualStudioSafeVersionsExtractor.cs

```diff
@@ -103,7 +104,9 @@ public static IEnumerable<Bundle> GetUninstallableBundles(IEnumerable<Bundle> bu
         {
             var (bundlesByDivisions, remainingBundles) = ApplyVersionDivisions(allBundles);
 
-            var bundlesAboveUpperLimit = remainingBundles.Where(bundle => bundle.Version.SemVer >= UpperLimit);
+            var bundlesAboveUpperLimit = CommandLineConfigs.CommandLineParseResult.ForceArgumentProvided()
```

Comment:
I don't think this does what you might expect? I think this just changes a string, which shouldn't really matter for uninstallation purposes. If the user specifies force (or however we decide they should be able to request uninstalling versions above the upper limit), I think we should instead add an extra version division [UpperLimit, infinity) and treat it the same as the rest.

Reply:
Thanks, this was left over from some of my initial changes and I forgot to back it out.

With respect to the extra version division suggestion, are you suggesting that we remove the "force argument provided" check around [CommandBundleFilter.FilterRequiredBundles](https://github.com/dotnet/cli-lab/pull/269/files#diff-bbb2c6b491062b0828246e408aee2723c174f1aa44318314f96886b47aec406dR71) and instead check if this param is provided and update the version groups created in VisualStudioSafeVersionsExtractor.ApplyWindowsVersionDivisions / VisualStudioSafeVersionsExtractor.ApplyMacVersionDivisions accordingly?

Reply:
Yes; that's what I was thinking. I haven't spent that much time with cli-lab, though, so it would probably be good to get signoff from someone else who's commented on this PR.